tools/lib/gzipTileset.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+var Cesium = require('cesium');
+var fsExtra = require('fs-extra');
+var path = require('path');
+var Promise = require('bluebird');
+var zlib = require('zlib');
+
+var fsExtraCopy = Promise.promisify(fsExtra.copy);
+var fsExtraEnsureDir = Promise.promisify(fsExtra.ensureDir);
+var fsExtraReadFile = Promise.promisify(fsExtra.readFile);
+
+var defaultValue = Cesium.defaultValue;
+var defined = Cesium.defined;
+var DeveloperError = Cesium.DeveloperError;
+
+module.exports = gzipTileset;
+
+/**
+ * gzips or gunzips the input tileset.
+ *
+ * @param {String} inputDirectory Path to the input directory.
+ * @param {Object} [outputDirectory] Path to the output directory.
+ * @param {Object} [options] Object with the following properties:
+ * @param {Boolean} [options.gzip=true] Whether to gzip or gunzip the tileset.
+ * @param {Boolean} [options.tilesOnly=false] Only gzip tiles, does not gzip tileset.json or other files.
+ * @param {Boolean} [options.verbose=false] If true prints out debug messages to the console.
+ */
+function gzipTileset(inputDirectory, outputDirectory, options) {
+    return new Promise(function(resolve, reject) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+        var gzip = defaultValue(options.gzip, true);
+        var tilesOnly = defaultValue(options.tilesOnly, false);
+        var verbose = defaultValue(options.verbose, false);
+
+        if (!defined(inputDirectory)) {
+            reject(new DeveloperError('inputDirectory is required'));
+        }
+        inputDirectory = path.normalize(inputDirectory);
```

Comment:
`path.normalize` for this and the next line really belong in the calling code, not here.